frontend/server/src/main/java/org/pytorch/serve/wlm/AsyncBatchAggregator.java

```diff
@@ -0,0 +1,196 @@
+                    job.response(
+                            prediction.getResp(),
+                            prediction.getContentType(),
+                            prediction.getStatusCode(),
+                            prediction.getReasonPhrase(),
+                            prediction.getHeaders());
+                } else {
+                    logger.warn(
+                            "Drop response for inference request {} due to client timeout",
+                            job.getPayload().getRequestId());
+                }
+                String streamNext =
+                        prediction
+                                .getHeaders()
+                                .get(org.pytorch.serve.util.messages.RequestInput.TS_STREAM_NEXT);
+                if ("false".equals(streamNext)) {
+                    jobs_in_backend.remove(jobId);
+                }
+            }
+
+        } else {
+            for (Map.Entry<String, Job> j : jobs_in_backend.entrySet()) {
+                if (j.getValue() == null) {
+                    throw new IllegalStateException(
+                            "Unexpected job in sendResponse() with non 200 status code: "
+                                    + j.getKey());
+                }
+                Job job = j.getValue();
+                if (job.getPayload().getClientExpireTS() > System.currentTimeMillis()) {
+                    job.sendError(message.getCode(), message.getMessage());
+                } else {
+                    logger.warn(
+                            "Drop error response for inference request {} due to client timeout",
+                            job.getPayload().getRequestId());
+                }
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void sendError(BaseModelRequest message, String error, int status) {
+        if (message instanceof ModelLoadModelRequest) {
+            logger.warn("Load model failed: {}, error: {}", message.getModelName(), error);
+            return;
+        }
+
+        if (message != null) {
+            ModelInferenceRequest msg = (ModelInferenceRequest) message;
+            for (RequestInput req : msg.getRequestBatch()) {
+                String requestId = req.getRequestId();
+                Job job = jobs_in_backend.remove(requestId);
+                if (job == null) {
+                    logger.error("Unexpected job in sendError(): " + requestId);
+                } else {
+                    job.sendError(status, error);
+                }
+            }
+            if (!jobs_in_backend.isEmpty()) {
+                // cleanJobs();
+                logger.error("Not all jobs got an error response.");
+            }
+        } else {
+            // Send the error message to all the jobs
+            for (Map.Entry<String, Job> j : jobs_in_backend.entrySet()) {
+                String jobsId = j.getValue().getJobId();
+                Job job = jobs_in_backend.get(jobsId);
```

Comment:
```suggestion
                Job job = jobs_in_backend.remove(jobsId);
```
Should we update state of `jobs_in_backend` here as well, like above?